test/TestServerProjects/httpInfrastructure/Generated/HttpSuccessRestClient.cs

```diff
@@ -668,29 +670,31 @@ internal HttpMessage CreateHead204Request()
 
         /// <summary> Return 204 status code if successful. </summary>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public async Task<Response> Head204Async(CancellationToken cancellationToken = default)
+        public async Task<Response<bool>> Head204Async(CancellationToken cancellationToken = default)
         {
             using var message = CreateHead204Request();
             await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
             switch (message.Response.Status)
             {
                 case 204:
```

Comment:
Because this switch is here you would only enter the branch on 204 response code, we need to get rid of it.